src/views/Swap/Swap.vue

```diff
@@ -12,11 +12,13 @@
       >
         Swap
       </NavBar>
-      <InfoNotification v-if="ethRequired">
-        <EthRequiredMessage :account-id="account.id"/>
+      <InfoNotification v-if="noTokens && sameNetwork">
```

Comment:
Why does something special need to be done for `sameNetwork`? 
Can you actually detail what behaviour you are trying to achieve? 
To my understanding: 
- Show {asset} balance required message when user swapping from {asset} and they don't have balance
- Show ETH balance required message when user swapping to ETH and they don't have balance